### DISCUSSION TITLE
Supabase Storage URL for local development

### DISCUSSION THREAD

**Discussion by mb21** [supabase/supabase#22432] [Questions] [answered]
What's the localhost URL I need to supply to tus-js-client when adapting [this example from the official docs](https://supabase.com/docs/guides/storage/uploads/resumable-uploads) to localhost?

What's the equivalent to `https://${SUPABASE_PROJECT_ID}.supabase.co/storage/v1/upload/resumable` on localhost? `http://localhost:54323/storage/v1/upload/resumable` gives a CORS error, so probably not the right endpoint. I see in the logs of the `supabase_storage_myproject` docker service "Server listening at http://0.0.0.0:5000", but how is that exposed via Docker and Kong?

Note: I'm using Supabase CLI with Docker Desktop on macOS, if it matters.

(P.S. I found https://github.com/orgs/supabase/discussions/7230, but no URL mentioned there.)

**Comment by devjume** [comment, marked as answer]
When you run `supabase status` command it gives following output for S3 Storage:
```
API URL: http://127.0.0.1:54321
GraphQL URL: http://127.0.0.1:54321/graphql/v1
S3 Storage URL: http://127.0.0.1:54321/storage/v1/s3 <---------- HERE
DB URL: postgresql://postgres:postgres@127.0.0.1:54322/postgres
Studio URL: http://127.0.0.1:54323
Inbucket URL: http://127.0.0.1:54324
```

Given that I think URL for resumable upload would be `http://localhost:54321/storage/v1/s3/upload/resumable`
Didn't test yet but I hope to hear if this helped.

Note: URL/port you tried using `http://localhost:54323` is for Supabase studio. Which is meant to viewed in browser. Not to interacted with code.

[Supabase CLI status command docs](https://supabase.com/docs/reference/cli/supabase-status?queryGroups=example&example=supabase-status-basic-usage)

**Reply by mb21** [reply]
Thanks for the answer! That's interesting, because for me (with `supabase` npm package `1.150.0`) the output of `supbase status` only contains:

```
supabase local development setup is running.

          DB URL: postgresql://postgres:postgres@127.0.0.1:54322/postgres
      Studio URL: http://127.0.0.1:54323
    Inbucket URL: http://127.0.0.1:54324
```

Hm.. and POSTing to both `http://localhost:54321/storage/v1/s3/upload/resumable` and `http://localhost:54321/storage/v1/upload/resumable` gives a 404.

**Reply by devjume** [reply]
Yea that's interesting indeed.

I am running Supabase locally with version `1.162.4`.  

And installed it with npm as [shown in docs](https://supabase.com/docs/guides/cli/getting-started?queryGroups=platform&platform=npx).
This runs Supabase locally with docker containers.

How did you install supabase in first place? Windows, mac, linux, npm?
When you run `docker ps` in terminal, what supabase related containers can you see?

**Reply by mb21** [reply]
Updating to `1.162.4` indeed fixed this, thanks!

**Reply by pruthvi2103** [reply]
Your containers were not fully downloaded in the initial setup. this is why you only see few containers. if this happens trying to delete all containers added by the cli can help too

**Reply by mb21** [reply]
@pruthvi2103 that's a known issue? that supabase cli sometimes just silently fails to download and run containers? But I saw `supabase_storage_myproject` running in Docker Desktop..?

**Reply by pruthvi2103** [reply]
not exactly a known issue, the cli likes to download the highest priority containers first then spins up the rest if its interrupted due to external causes like user interrupt, internet being bad or docker itself then it fails silently
